endpoints/openrtb2/video_auction.go

```diff
@@ -617,7 +618,18 @@ func (deps *endpointDeps) parseVideoRequest(request []byte, headers http.Header)
 
 	//if Device.UA is not present in request body, init it with user-agent from request header if it's present
 	if req.Device.UA == "" {
-		req.Device.UA = headers.Get("User-Agent")
+		ua := headers.Get("User-Agent")
+
+		//Check UA is encoded. Without it the `+` character would get changed to a space if not actually encoded
+		if strings.ContainsAny(ua, "%") {
+			var err error
+			req.Device.UA, err = url.QueryUnescape(ua)
```

Comment:
This feels a little clumsy. Should we assign the `url.QueryUnescape()` result to a variable, and then assign it to `req.Device.UA` after we check `err == nil`? I am open on this question, can't quite envision a sequence that is definitely "not clumsy".

Reply:
We just did this way to avoid having to create a 3rd variable for it if we need to set it back to the original value for the error case, unless QueryUnescape() returns the original string in an error case then we can go back to using ua there and always set it (without needing to check the error in that case since either way ua would be what we want to use).

Reply:
`req.Device.UA` needs to be assigned in any case. We're copying the UA from the http headers to the OpenRTB request. I thought it looked fine. I'm also fine if you want to use an intermediate variable.

It's not that we're changing `req.Device.UA` to possibly nil and then back to ua. It's nil from the start.

Reply:
leave it it as is then. It looks clumsy to my eyes, but I couldn't find a better suggestion, so I opened it up to see if anyone else could see one.

Reply:
Can always think on it and iterate in the future.